espei/optimizers/opt_mcmc.py

```diff
@@ -167,14 +166,27 @@ def save_sampler_state(self):
     def do_sampling(self, chains, iterations):
         progbar_width = 30
         _log.info('Running MCMC for %s iterations.', iterations)
+        iter_start_time = time.time()
         try:
             for i, result in enumerate(self.sampler.sample(chains, iterations=iterations)):
                 # progress bar
                 if (i + 1) % self.save_interval == 0:
                     self.save_sampler_state()
                     _log.trace('Acceptance ratios for parameters: %s', self.sampler.acceptance_fraction)
                 n = int((progbar_width) * float(i + 1) / iterations)
-                _log.info("\r[%s%s] (%d of %d)\n", '#' * n, ' ' * (progbar_width - n), i + 1, iterations)
+                _log.info("\r[%s%s] (%d of %d)", '#' * n, ' ' * (progbar_width - n), i + 1, iterations)
+                if self.scheduler is not None:
+                    iter_curr_time = time.time()
+                    scheduler_info = self.scheduler.scheduler_info()
+                    memory = []
+                    for w in scheduler_info['workers']:
+                        memory.append(float(scheduler_info['workers'][w]['metrics'].get('memory', 0)))
+                    _log.info('Current time (s): %.1f, Total memory (GB): %.3f, Average worker memory (GB): %.3f\n', iter_curr_time - iter_start_time, np.sum(memory)/1e9, np.average(memory)/1e9)
+                    
+                    workers = list(self.scheduler.has_what().keys())
+                    if i % 3 == 0:
+                        self.scheduler.restart_workers([workers[0], workers[1], workers[4], workers[5]])
+                    self.wrapper.check_workers_have_futures()
```

Comment:
Can you special case this for dask? The code is otherwise written to be somewhat generic in taking a scheduler as long as it has a `map()` method and I'd like to keep that behavior.

Reply:
As in also check if the scheduler is an instance of a dask.Client?

Reply:
Yeah, that would work. We want our dask specific mitigations, but to continue allowing non-dask scheduler usage (if anyone is using)